File: src/org/foonugget/kdict/ui/SearchHistoryListAdapter.java

package org.foonugget.kdict.ui;

import java.util.List;

import android.content.Context;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.google.inject.Inject;
import com.google.inject.assistedinject.Assisted;

public class SearchHistoryListAdapter extends BaseAdapter {
    @SuppressWarnings("unused")
    private final static String TAG = SearchHistoryListAdapter.class.getSimpleName();

    private final Context mContext;

    private List<String> mHistory;

    private OnClickListener mClickSearchHistoryListener;

    /**
     * Creation factory
     */
    public interface Factory {
        SearchHistoryListAdapter create(List<String> matchesList, OnClickListener clickListener);
    }

    @Inject
    public SearchHistoryListAdapter(Context context, @Assisted
    List<String> historyList,
            @Assisted
            OnClickListener clickSearchHistoryEntry) {
        mContext = context;
        mHistory = historyList;
        mClickSearchHistoryListener = clickSearchHistoryEntry;

        if (mHistory == null || mHistory.size() == 0) {
            mHistory.add("");
        }
    }

    @Override
    public int getCount() {
        return mHistory.size();
    }

    @Override
    public String getItem(int position) {
        return mHistory.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        final String searchString = getItem(position);
        TextView textView = new TextView(mContext);
        textView.setText(searchString);
        textView.setTextSize(30);
        textView.setOnClickListener(mClickSearchHistoryListener);
        return textView;
    }

}

File: app/src/main/java/org/foonugget/kdict/ui/SearchHistoryListAdapter.kt
package org.foonugget.kdict.ui

import android.content.Context
import android.view.View
import android.view.View.OnClickListener
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

class SearchHistoryListAdapter(
    private val context: Context,
    private val historyList: List<String>,
    private val clickSearchHistoryEntry: OnClickListener
) : BaseAdapter() {

    override fun getCount(): Int = historyList.size

    override fun getItem(position: Int): String = historyList[position]

    override fun getItemId(position: Int): Long = position.toLong()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val searchString = getItem(position)
        val textView = TextView(context)
        textView.text = searchString
        textView.textSize = TEXT_SIZE
        textView.setOnClickListener(clickSearchHistoryEntry)
        return textView
    }

    class Factory(private val context: Context) {
        fun create(
            historyList: List<String>,
            clickListener: OnClickListener
        ): SearchHistoryListAdapter {
            return SearchHistoryListAdapter(context, historyList, clickListener)
        }
    }

    companion object {
        private const val TEXT_SIZE = 30f
    }
}
